src/hooks/useTransactionSteps.ts

```diff
@@ -77,12 +93,12 @@ const useTransactionSteps = ({ transaction, onComplete }: Props) => {
     const sourceTransactionFinalized = stepEvaluator(transaction.block, bestBlockFinalized);
     const blockFinalityRelayed = stepEvaluator(transaction.block, bestBridgedFinalizedBlockOnTarget);
     const messageDelivered = stepEvaluator(transaction.messageNonce, latestReceivedNonceRuntimeApi);
-    const messageFinalizedOnTarget = stepEvaluator(transaction.messageNonce, targetMessageDeliveryBlock);
+    const messageFinalizedOnTarget = stepEvaluator(transactionNonceOfTargetFinalizedBlock, nonceOfCurrentTargetBlock);
     const sourceConfirmationReceived = stepEvaluator(transaction.messageNonce, latestReceivedNonceOnSource);
     const onChainCompleted = (value: boolean) => completionStatus(value) === TransactionStatusEnum.COMPLETED;
 
     if (messageDelivered && !targetMessageDeliveryBlock) {
-      setTargetMessageDeliveryBlock(bestBridgedFinalizedBlockOnTarget);
+      setTargetMessageDeliveryBlock(bestBlockFinalizedOnTarget);
```

Comment:
1) For some reason i was setting the bridged block of the target chain , what i needed was the `bestBlockFinalizedOnTarget`